### PR TITLE
build.js: support outputting esbuild metadata file

### PR DESCRIPTION
**Pull request by jelly**
ESbuild supports outputting a metadata file which can be used to view bundle size information using https://esbuild.github.io/analyze/

---

Happy to propagate this along all Cockpit projects later :)

image

Can't say if treeshaking is working, because we seem to include all PF css for some reason :thinking: 

image

You can click on some CSS which should not be there and see why it was imported:

image

Another example:

```
[jelle@t14s][~/projects/cockpit-files]%git grep Drawer
[jelle@t14s][~/projects/cockpit-files]%
```

image

### CODE REVIEW

**Comment by jelly** [comment]
This whole things reminds me of barrel files in JavaScript for imports, which I really never investigated but was something funky regarding imports and running `oxlint`'s anaylzer:

https://github.com/oxc-project/oxc/releases/tag/oxlint_v0.4.0



```
  x oxc(no-barrel-file): Barrel file detected, 145 modules are loaded.
   ,-[packages/react-core/src/components/Alert/index.ts:1:15]
 1 | export * from './Alert';
   :               ^^^^|^^^^
   :                   `-- 53 modules
 2 | export * from './AlertContext';
   :               ^^^^^^^^|^^^^^^^
   :                       `-- 1 modules
 3 | export * from './AlertActionCloseButton';
   :               ^^^^^^^^^^^^^|^^^^^^^^^^^^
   :                            `-- 17 modules
 4 | export * from './AlertActionLink';
   :               ^^^^^^^^^|^^^^^^^^^
   :                        `-- 13 modules
 5 | export * from './AlertGroup';
   :               ^^^^^^^|^^^^^^
   :                      `-- 61 modules
   `----
  help: Loading 145 modules is slow for runtimes and bundlers.
        The configured threshold is 100.
        See also: <https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7>.
```

**Comment by jelly** [comment]
So as side effect we discover that we import things wrong in Cockpit:
```
[jelle@t14s][~/projects/cockpit/main]%git grep Tabs
pkg/lib/cockpit-components-listing-panel.jsx:import { Tab, TabTitleText, Tabs } from "@patternfly/react-core/dist/esm/components/Tabs/index.js";
```

```
[jelle@t14s][~/projects/cockpit/main]%cat -p node_modules/@patternfly/react-core/dist/esm/components/Tabs/index.js
export * from './Tab';
export * from './TabAction';
export * from './Tabs';
export * from './TabContent';
export * from './TabContentBody';
export * from './TabsContext';
export * from './TabTitleText';
export * from './TabTitleIcon';
//# sourceMappingURL=index.js.map
```

So we import Tab,TabAction, etc. while we only want Tab, TabTitle, Tabs

And yeah `oxlint` agrees:

```
[jelle@t14s][~/projects/cockpit/main]%oxlint --import-plugin -D no-barrel-file node_modules/@patternfly/react-core/dist/esm/components/Tabs/index.js

  × oxc(no-barrel-file): Barrel file detected, 446 modules are loaded.
   ╭─[node_modules/@patternfly/react-core/dist/esm/components/Tabs/index.js:1:15]
 1 │ export * from './Tab';
   ·               ───┬───
   ·                  ╰── 108 modules
 2 │ export * from './TabAction';
   ·               ──────┬──────
   ·                     ╰── 97 modules
 3 │ export * from './Tabs';
   ·               ────┬───
   ·                   ╰── 140 modules
 4 │ export * from './TabContent';
   ·               ───────┬──────
   ·                      ╰── 88 modules
 5 │ export * from './TabContentBody';
   ·               ─────────┬────────
   ·                        ╰── 4 modules
 6 │ export * from './TabsContext';
   ·               ───────┬───────
   ·                      ╰── 1 modules
 7 │ export * from './TabTitleText';
   ·               ────────┬───────
   ·                       ╰── 4 modules
 8 │ export * from './TabTitleIcon';
   ·               ────────┬───────
   ·                       ╰── 4 modules
 9 │ //# sourceMappingURL=index.js.map
   ╰────
  help: Loading 446 modules is slow for runtimes and bundlers.
        The configured threshold is 100.
        See also: <https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7>.

Finished in 13ms on 1 file with 1 rules using 8 threads.
Found 0 warnings and 1 error.
```

File: build.js [hunk 1]
```diff
@@ -90,6 +91,7 @@ const context = await esbuild.context({
     minify: production,
     nodePaths,
     outdir,
+    ...(args.metafile && { metafile: true }),
```
Comment (allisonkarlitskaya) [inline build.js:94]: This is overly complicated.  Just let it be set to false...

Comment (allisonkarlitskaya) [reply build.js:94]: `metafile: !!args.metafile`?

Comment (jelly) [reply build.js]: Done

**Review by allisonkarlitskaya — APPROVED** [review]
kthxbye

**Pull request merged**